homeassistant/components/knx/light.py

```diff
@@ -329,29 +346,19 @@ def supported_color_modes(self) -> set | None:
 
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the light on."""
-        # ignore arguments if not supported to fall back to set_on()
-        brightness = (
-            kwargs.get(ATTR_BRIGHTNESS)
-            if self._device.supports_brightness
-            or self.color_mode in (COLOR_MODE_RGB, COLOR_MODE_RGBW)
-            else None
-        )
-        mireds = (
-            kwargs.get(ATTR_COLOR_TEMP)
-            if self.color_mode == COLOR_MODE_COLOR_TEMP
-            else None
-        )
-        rgb = kwargs.get(ATTR_RGB_COLOR) if self.color_mode == COLOR_MODE_RGB else None
-        rgbw = (
-            kwargs.get(ATTR_RGBW_COLOR) if self.color_mode == COLOR_MODE_RGBW else None
-        )
```

Comment:
removed the checks for supported modes because of https://github.com/home-assistant/core/pull/51084